packages/vscode-extension/src/webview/hooks/useDependencyErrors.ts

```diff
@@ -1,102 +1,33 @@
+import { use$ } from "@legendapp/state/react";
+import { useMemo } from "react";
+import { useStore } from "../providers/storeProvider";
 import {
-  PropsWithChildren,
-  createContext,
-  useCallback,
-  useContext,
-  useEffect,
-  useMemo,
-  useState,
-} from "react";
-import { makeProxy } from "../utilities/rpc";
-import {
-  DependencyManagerInterface,
-  DependencyStatus,
-  MinSupportedVersion,
-} from "../../common/DependencyManager";
-
-const dependencyManager = makeProxy<DependencyManagerInterface>("DependencyManager");
+  ApplicationDependency,
+  ApplicationDependencyRecord,
+  EnvironmentDependency,
+  EnvironmentDependencyRecord,
+} from "../../common/State";
+import { MinSupportedVersion } from "../../common/Constants";
 
-type Dependency =
-  | "nodejs"
-  | "packageManager"
-  | "androidEmulator"
-  | "xcode"
-  | "cocoaPods"
-  | "nodeModules"
-  | "ios"
-  | "android"
-  | "pods"
-  | "reactNative"
-  | "expo"
-  | "expoRouter"
-  | "storybook"
-  | "easCli";
-
-type ErrorType = "ios" | "simulator" | "emulator" | "android" | "common";
 type Errors = Partial<Record<ErrorType, { message: string }>>;
-type DependencyRecord = Partial<Record<Dependency, DependencyStatus>>;
-
-interface DependenciesContextProps {
-  dependencies: DependencyRecord;
-  errors: Errors | undefined;
-  runDiagnostics: () => Promise<void>;
-}
-
-const DependenciesContext = createContext<DependenciesContextProps>({
-  dependencies: {},
-  errors: undefined,
-  runDiagnostics: () => {
-    throw new Error("Provider not initialized");
-  },
-});
-
-export default function DependenciesProvider({ children }: PropsWithChildren) {
-  const [depsState, updateDepsState] = useState<DependencyRecord>({});
-
-  const runDiagnostics = useCallback(() => {
-    return dependencyManager.runAllDependencyChecks();
-  }, []);
-
-  useEffect(() => {
-    const dependencies: DependencyRecord = {};
-
-    function handleUpdatedDependency(dependency: Dependency, status: DependencyStatus) {
-      dependencies[dependency] = status;
-      updateDepsState({ ...dependencies });
-    }
-
-    dependencyManager.addListener(handleUpdatedDependency);
-    runDiagnostics();
-
-    return () => {
-      dependencyManager.removeListener(handleUpdatedDependency);
-    };
-  }, []);
-
-  const contextValue = useMemo(() => {
-    return {
-      dependencies: depsState,
-      runDiagnostics,
-      errors: getErrors(depsState),
-    };
-  }, [depsState, runDiagnostics, getErrors]);
-
-  return (
-    <DependenciesContext.Provider value={contextValue}>{children}</DependenciesContext.Provider>
+type ErrorType = "ios" | "simulator" | "emulator" | "android" | "common";
+type Dependency = ApplicationDependency | EnvironmentDependency;
+type DependencyRecord = ApplicationDependencyRecord | EnvironmentDependencyRecord;
```

Comment:
Same comment about the type name as for `ApplicationDependencyRecord`, `EnvironmentDependencyRecord`